src/domain/request/requestExecutor/UrlRequestExecutor.ts

```diff
@@ -0,0 +1,31 @@
+import {LoggerPort} from '@core/domain/common/port';
+import RequestExecutor from '@core/domain/common/port/RequestExecutorPort';
+import FetchDataUseCase from '@core/domain/common/usecase/FetchDataUseCase';
+import SelectDataUseCase from '@core/domain/common/usecase/SelectDataUseCase';
+import Request from '@core/domain/request/Request';
+import Response from '@core/domain/response/Response';
+
+class UrlRequestExecutor implements RequestExecutor {
```

Comment:
Why does RequestExecutor know that it is handling url based request? Maybe moving this knowledge to fetchDataUseCase would be cleaner approach?

Reply:
It is indeed implemented like this. All url-specific stuff happens in Fetcher.

UrlRequestExecutor was just my attemp of naming a group of `['json', 'xml', 'html', 'ipfs']` dataSources. Do you have an idea for a better name?